db/compaction/compaction_job.cc

```diff
@@ -1518,14 +1021,19 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
       c_iter->ResetRecordCounts();
       RecordCompactionIOStats();
     }
-
+    // Only enable preclude_bottommost feature when the bottommost output level
+    // > 1
+    if (bottommost_level_ && compact_->compaction->output_level() > 1) {
```

Comment:
It feels that the iter->preclude_bottommost() check is likely to be false than the other two. Can we first check that one first?